packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -493,13 +544,21 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// [cameraId] is not used.
   @override
   Future<XFile> takePicture(int cameraId) async {
+    // Set flash mode.
     if (_currentFlashMode != null) {
       await imageCapture!.setFlashMode(_currentFlashMode!);
     } else if (torchEnabled) {
       // Ensure any previously set flash modes are unset when torch mode has
       // been enabled.
       await imageCapture!.setFlashMode(ImageCapture.flashModeOff);
     }
+
+    // Set target rotation to current photo orientation only if capture
+    // orientation not locked.
+    if (!captureOrientationLocked && shouldSetDefaultRotation) {
+      await imageCapture!.setTargetRotation(await proxy.getDefaultRotation());
```

Comment:
In testing, does this achieve the same behavior between "plugin-managed" unlocked case (i.e. lock->unluck), and the "camerax" managed unlocked case? I'm curious because after reading the documentation on `ImageCapture`'s [`setTargetRotation`](https://developer.android.com/reference/androidx/camera/core/ImageCapture#setTargetRotation(int)), particularly:
> If no target rotation is set by the application, it is set to the value of [getRotation](https://developer.android.com/reference/android/view/Display.html#getRotation()) of the default display at the time the use case is bound.

To me this sounds like we would be doing "rotation of default display at time of photo", while camerax is doing "rotation of default display at time of use case creation", which sounds like it would be different if the phone was rotated after creation. Am I misinterpreting the docs, or do the docs not line up with what happens when we test the use cases?

(this comment applies to how we manage the rotation for all use cases, I just put it here as an example)

Reply:
My understanding is that the default display does not change anywhere in the plugin and the rotation of that display always will return the rotation needed to return to its "natural" orientation (https://developer.android.com/reference/android/view/Display.html#getRotation()). From my testing, this yields the desired result of matching `camera_android` behavior.

This makes sense to me because my interpretation of "To return to the default value, set the value to `context.getSystemService(WindowManager.class).getDefaultDisplay().getRotation();`" was that you should be able to do that at anytime to have the `UseCase` return to its default rotation by CameraX standards. I think there would be a concern here if the default display were changing between binding `UseCase`s and setting the target rotation, but like I mentioned, I don't believe it should be a concern of ours.